### ISSUE TITLE
Underscored and non-underscored type name collision

### ISSUE THREAD

**Issue by tomaszwojcik** [rmosolgo/graphql-ruby#3249]
**Is your feature request related to a problem? Please describe.**

The schema I get from GraphQL server contains underscored and non-underscored identifiers like `_Product` and `Product` which leads to `ArgumentError: Can't define Product to represent type Product because it's already defined`.

**Describe the solution you'd like**

I would like to be able to work with servers that provide such schemas - so I need some way to avoid name collision in that case.

**Describe alternatives you've considered**

After skimming the docs I haven't found any option to handle such weird names.

Quick hack with changing `normalize_type_name` in `client/schema.rb` to always return `type_name` ends up with `NameError: wrong constant name _Product`.

**Additional context**

Code I use to load the file:
```
Schema = GraphQL::Schema.from_definition('absolute_path_to_dir/schema.graphql')
```

Problematic schema file:
[schema.graphql.zip](https://github.com/rmosolgo/graphql-ruby/files/5648659/schema.graphql.zip)

**Comment by rmosolgo** [comment]
Hi, could you please share the full backtrace of the error you encountered? I don't recognize that error message. 

For me, loading that schema works fine: 

```ruby 
~/code/graphql-ruby % irb
irb(main):001:0> require "graphql"
# => true
irb(main):002:0> schema = GraphQL::Schema.from_definition("/Users/rmosolgo/Downloads/schema.graphql")
# => #<Class:0x00007fe24b1b8430>
irb(main):003:0> schema.get_type("_Product")
# => #<Class:0x00007fe24b281fd8>
irb(main):004:0> schema.get_type("Product")
# => #<Class:0x00007fe24b240498>
```

**Comment by tomaszwojcik** [comment]
Unofortunately I removed the gem that uses `graphql` from Rails project - `graphql-client` and I'm not able to reproduce the issue because I've installed the latest `graphql` manually (1.11.6) and then installed `graphql-client` again and that seems to fix the problem.

I guess that it must have been issue with `graphql-client` (which probably installed older version of `graphql` dependency). Weird, but I'm glad it works.

Thanks for quick response!
